app/gui2/ydoc-server/edits.ts

```diff
@@ -116,12 +116,34 @@ export function translateVisualizationFromFile(
   }
 }
 
+function stupidDiff(oldString: string, newString: string): diff.Diff[] {
+  const minLength = Math.min(oldString.length, newString.length)
+  let commonPrefixLen, commonSuffixLen
+  for (commonPrefixLen = 0; commonPrefixLen < minLength; ++commonPrefixLen)
+    if (oldString[commonPrefixLen] !== newString[commonPrefixLen]) break
+  if (oldString.length === newString.length && oldString.length === commonPrefixLen)
+    return [[0, oldString]]
+  for (commonSuffixLen = 0; commonSuffixLen < minLength - commonPrefixLen; ++commonSuffixLen)
+    if (oldString.at(-1 - commonSuffixLen) !== newString.at(-1 - commonSuffixLen)) break
+  const commonPrefix = oldString.substring(0, commonPrefixLen)
+  const removed = oldString.substring(commonPrefixLen, oldString.length - commonSuffixLen)
+  const added = newString.substring(commonPrefixLen, newString.length - commonSuffixLen)
+  const commonSuffix = oldString.substring(oldString.length - commonSuffixLen, oldString.length)
+  return (commonPrefix ? ([[0, commonPrefix]] as diff.Diff[]) : [])
+    .concat(removed ? [[-1, removed]] : [])
+    .concat(added ? [[1, added]] : [])
+    .concat(commonSuffix ? [[0, commonSuffix]] : [])
+}
+
 export function applyDiffAsTextEdits(
   lineOffset: number,
   oldString: string,
   newString: string,
 ): TextEdit[] {
-  const changes = diff(oldString, newString)
+  const changes =
+    oldString.length + newString.length > 50000
```

Comment:
how was that threshold number chosen? also let's move it to a named and documented constant